Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/Beckn/Select.hs

```diff
@@ -164,4 +191,14 @@ buildSearchReqLocation merchantId sessionToken _ latLong@Maps.LatLong {..} = do
   now <- getCurrentTime
   let createdAt = now
       updatedAt = now
-  pure DLoc.SearchReqLocation {..}
+  pure DLoc.SearchReqLocation {lat = latLong.lat, lon = latLong.lon, ..}
```

Comment:
this also can be refactored a little to avoid duplication

```
buildSearchReqLocation :: (EncFlow m r, CacheFlow m r, EsqDBFlow m r, CoreMetrics m) => Id DM.Merchant -> Text -> Maybe BA.Address -> Maybe Maps.Language -> LatLong -> m DLoc.SearchReqLocation
buildSearchReqLocation merchantId sessionToken address customerLanguage latLong@Maps.LatLong {..} = do
  Address {..} <- case address of
    Just loc | customerLanguage == Just Maps.ENGLISH ->
      pure $ Address
        { areaCode = loc.area_code,
          street = loc.street,
          city = loc.city,
          state = loc.state,
          country = loc.country,
          building = loc.building,
          area = loc.ward,
          full_address = decodeAddress loc
        }
    _ -> getAddressByGetPlaceName merchantId sessionToken latLong
  id <- Id <$> generateGUID
  now <- getCurrentTime
  let createdAt = now
      updatedAt = now
  pure DLoc.SearchReqLocation {..}

getAddressByGetPlaceName :: (EncFlow m r, CacheFlow m r, EsqDBFlow m r, CoreMetrics m) => Id DM.Merchant -> Text -> LatLong -> m Address
getAddressByGetPlaceName merchantId sessionToken latLong = do
  pickupRes <-
    Maps.getPlaceName merchantId $
      Maps.GetPlaceNameReq
        { getBy = Maps.ByLatLong latLong,
          sessionToken = Just sessionToken,
          language = Nothing
        }
  mkLocation pickupRes
```

Reply:
Sure